core/src/main/scala/com/metamx/tranquility/config/ConfigHelper.scala

```diff
@@ -0,0 +1,125 @@
+package com.metamx.tranquility.config
+
+import com.metamx.common.scala.Predef._
+import com.metamx.common.scala.Yaml
+import com.metamx.common.scala.untyped._
+import com.metamx.tranquility.druid.DruidBeams
+import com.metamx.tranquility.druid.DruidEnvironment
+import com.metamx.tranquility.druid.DruidGuicer
+import com.metamx.tranquility.druid.DruidLocation
+import com.metamx.tranquility.finagle.FinagleRegistry
+import com.metamx.tranquility.tranquilizer.Tranquilizer
+import io.druid.segment.realtime.FireDepartment
+import java.io.InputStream
+import java.util.Properties
+import org.apache.curator.framework.CuratorFramework
+import org.skife.config.ConfigurationObjectFactory
+
+/**
+ * Helper methods to perform shared tasks based on common configuration file semantics.
+ */
+object ConfigHelper
+{
+  /**
+   * Builds configuration object from YAML file.
+   *
+   * This method may have the side effect of modifying system properties if the YAML file contains properties beginning
+   * with "druid.extensions." - this is necessary as DruidGuicer reads the system properties to know what extensions
+   * need to be loaded. If the properties have already been set elsewhere, the system properties will not be modified.
+   */
+  def readConfigYaml[T <: TranquilityConfig](
+    in: InputStream,
+    clazz: Class[T]
+    ): (T, Map[String, DataSourceConfig[T]], Dict) =
+  {
+    val yaml = in.withFinally(_.close()) { in =>
+      dict(Yaml.load(in))
+    }
+    val globalProperties: Dict = getAsDict(yaml, "properties")
+    val globalConfig = mapConfig(globalProperties, clazz)
+
+    for (prop <- globalProperties.filterKeys {
```

Comment:
generally prefer to format this on one line as `globalProperties.filterKeys(_.startsWith("druid.extensions."))`